solarforecastarbiter/pvmodel.py

```diff
@@ -0,0 +1,281 @@
+"""
+Calculate AC power and modeling intermediates from system metadata,
+times, and weather data.
+
+Steps are:
+
+1. Calculate solar position using solar_position
+2. If not already known, calculate 3 irradiance components from measured
+   GHI using irradiance_components or modeled clear sky using clearsky.
+3. calculate_poa_effective
+4. calculate_power
+
+See :py:func:`ghi_to_power` and :py:func:`ghi_dni_dhi_to_power`
+"""
+
+import pvlib
+
+from solarforecastarbiter import datamodel
+
+
+def calculate_solar_position(site, times):
+    """
+    Calculates solar position using pvlib's implementation of NREL SPA.
+
+    Parameters
+    ----------
+    site : datamodel.Site
+    times : pd.DatetimeIndex
+
+    Returns
+    -------
+    solar_position : pd.DataFrame
+        The DataFrame will have the following columns: apparent_zenith
+        (degrees), zenith (degrees), apparent_elevation (degrees),
+        elevation (degrees), azimuth (degrees),
+        equation_of_time (minutes).
```

Comment:
:man_shrugging:  about df or components but at least should enforce that these are the columns

Reply:
we're pinning the versions and I am sure this will not change in pvlib over the time frame we care about.